target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-mf289f.dts

```diff
@@ -0,0 +1,435 @@
+                        status = "okay";
+                };
+
+                tcsr@1949000 {
+                        compatible = "qcom,tcsr";
+                        reg = <0x1949000 0x100>;
+                        qcom,wifi_glb_cfg = <TCSR_WIFI_GLB_CFG>;
+                };
+
+                tcsr@194b000 {
+                        /* select hostmode */
+                        compatible = "qcom,tcsr";
+                        reg = <0x194b000 0x100>;
+                        qcom,usb-hsphy-mode-select = <TCSR_USB_HSPHY_HOST_MODE>;
+                        status = "okay";
+                };
+
+                ess_tcsr@1953000 {
+                        compatible = "qcom,tcsr";
+                        reg = <0x1953000 0x1000>;
+                        qcom,ess-interface-select = <TCSR_ESS_PSGMII>;
+                };
+
+                tcsr@1957000 {
+                        compatible = "qcom,tcsr";
+                        reg = <0x1957000 0x100>;
+                        qcom,wifi_noc_memtype_m0_m2 = <TCSR_WIFI_NOC_MEMTYPE_M0_M2>;
+                };
+
+                ess-switch@c000000 {
+                        status = "okay";
+                };
+
+                edma@c080000 {
+                        status = "okay";
+                };
+        };
+};
+
+&prng {
+        status = "okay";
+};
+
+&mdio {
+        status = "okay";
+        pinctrl-0 = <&mdio_pins>;
+        pinctrl-names = "default";
+        reset-gpios = <&tlmm 47 GPIO_ACTIVE_LOW>;
+        reset-delay-us = <2000>;
+};
+
+&watchdog {
+        status = "okay";
+};
+
+&blsp_dma {
+        status = "okay";
+};
+
+&usb2 {
+        status = "okay";
+};
+
+&usb3 {
+        status = "okay";
+};
+
+&blsp1_spi1 {
+        pinctrl-0 = <&spi_0_pins>;
+        pinctrl-names = "default";
+        status = "okay";
+        cs-gpios = <&tlmm 12 GPIO_ACTIVE_HIGH>,
+                   <&tlmm 54 GPIO_ACTIVE_HIGH>;
+
+        flash@0 {
+                compatible = "jedec,spi-nor";
+                #address-cells = <1>;
+                #size-cells = <1>;
+                reg = <0>;
+                spi-max-frequency = <24000000>;
+
+                partitions {
+                        compatible = "fixed-partitions";
+                        #address-cells = <1>;
+                        #size-cells = <1>;
+
+                        partition@0 {
+                                label = "0:SBL1";
+                                reg = <0x0 0x40000>;
+                                read-only;
+                        };
+
+                        partition@40000 {
+                                label = "0:MIBIB";
+                                reg = <0x40000 0x20000>;
+                                read-only;
+                        };
+
+                        partition@60000 {
+                                label = "0:QSEE";
+                                reg = <0x60000 0x60000>;
+                                read-only;
+                                };

```

Comment:
Please use indentations instead of spaces in this file, and pay attention to the alignment, blank lines.

Reply:
fixed, please refer to new PR: https://github.com/openwrt/openwrt/pull/10708